torch_ttnn/backend.py

```diff
@@ -122,6 +135,48 @@ def aten_backend(
     gm.graph.lint()
     gm.recompile()
 
+    # Get the memory manager object for memory analysis
+    if option.run_mem_analysis:
+        option._memory_manager = mem_pass.memory_manager
+
+    # Run eviction opt pass if enabled
+    if option.run_eviction_opt == True:
+        assert (
+            option.run_mem_analysis == True
+        ), "Eviction pass depends on memory analysis pass!"
+        from torch_ttnn.passes.eviction_pass import EvictionPass
+
+        mm = option._memory_manager
+        nth_eviction = 1
+        # See if SRAM overflows and eviction is required
+        while mem_utils.check_sram_overflow(mm) is True:
```

Comment:
`While not true` loops often tend to go overboard. In this case user might think that compiler hanged.
A better approach is to set the limit and when the limit is hit - to tell a user about that. Something like
```
Tensor eviction step hit the iteration limit. Please set compiler option `Options.eviction_timelimit=` or `Options.eviction_steps_limit=..` to increase the timelimit. To disable the limit set `sys.maxint`.
```

Reply:
The users don't know anything about how many evictions are required to fit the model in memory. That comes from memory analysis and not every eviction would be successful. I think the current while loop could end up as an infinite loop if eviction fails. So I am thinking of finding out how many evictions are possible and then if the loop runs that many times and still doesn't fit in memory, then eviction strategy couldn't fit the model in memory and throws an assert.

Reply:
Thats the point in general. User does not know, but you can't cause an infinite loop here. Make sure you don't cause an infinite loop.